src/org/omegat/util/StringUtil.java

```diff
@@ -511,4 +511,31 @@ public static String format(String str, Object... arguments) {
         str = str.replaceAll("'", "''");
         return MessageFormat.format(str, arguments);
     }
+
+    /**
+     * convert Zenkaku/Hankaku alphabet to regex  that match both
+     *
+     * replace (\uFF01 - \uFF5E) to (\u0021- \u007E)
+     * such as text = text.replaceAll("u\0021", "(\uFF01|\u0021)");
+     * except for chars that consist of regex.
+     * FIXME: handling 0x20-0x2f, 0x5c
+     *
+     * @param text
+     * @return String with regex
+     */
+    public static String replaceZenHan(String text) {
```

Comment:
I don't think this is a good way to go about matching character width for search purposes. Building a regex is unnecessarily complex and not reusable. I haven't benchmarked, but it smells unperformant as well. Also you are not addressing hangul at all.

Instead of this, I suggest the following algorithm:
1. Use a new util method `boolean containsFullWidth(String)` to check if the comparison text needs normalization at all. In the vast majority of cases (i.e. not Japanese text) this will return `false`.
2. If above returns `true`, normalize the comparison text with a new util method `String toHalfWidth(String)`.

I would base the implementation of both of the above on Okapi's [FullWidthConversionStep](https://bitbucket.org/okapiframework/okapi/src/8aa39fcb9977e5cf8e96b566bddbdfc89efa78b5/okapi/steps/fullwidthconversion/src/main/java/net/sf/okapi/steps/fullwidthconversion/FullWidthConversionStep.java?at=master&fileviewer=file-view-default#FullWidthConversionStep.java-104). It is, as far as I know, comprehensive in its handling of full-width/half-width characters.